vir/src/data.rs

```diff
@@ -96,6 +96,7 @@ impl From<&mir::BinOp> for BinOpKind {
 
 #[derive(PartialEq, Eq, Clone, Copy, Serialize, Deserialize, Hash)]
 pub enum ConstData {
+    Perm(i128, i128),
```

Comment:
Maybe I missed it, but I think the PR does not make use of fractional permissions anywhere. If so, please remove them.